src/dbnode/persist/fs/seek_manager_test.go

```diff
@@ -680,3 +680,34 @@ func TestSeekerManagerCacheShardIndicesSkipNotFound(t *testing.T) {
 
 	require.NoError(t, m.Close())
 }
+
+func TestSeekerManagerReturnShard(t *testing.T) {
+	defer leaktest.CheckTimeout(t, 1*time.Minute)()
+
+	shards := []uint32{2, 5}
+	metadata := testNs1Metadata(t)
+	shardSet, err := sharding.NewShardSet(
+		sharding.NewShards(shards, shard.Available),
+		sharding.DefaultHashFn(1),
+	)
+	require.NoError(t, err)
+	m := NewSeekerManager(nil, testDefaultOpts, defaultTestBlockRetrieverOptions).(*seekerManager)
+	require.NoError(t, m.Open(metadata, shardSet))
+	defer require.NoError(t, m.Close())
```

Comment:
Hmm, I think the arguments to deferred call are evaluated immediately, which would close `seekerManager` right away.

Reply:
Good catch, thanks